doc/N-API Basic Types.md

```diff
@@ -0,0 +1,373 @@
+# N-API Documentation
```

Comment:
A general thought is that some intro at the top that introduces the basic types and how they are used might be useful before the section with the details.  We can add that later though so what you have here would be good to land without that.